### ISSUE TITLE
error: cannot operate on output 'out' of the unbuilt derivation

### ISSUE THREAD

**Issue by DavHau** [risicle/nix-heuristic-gc#7]
Using the latest version from the flake of this repo, I get this error:

```
finding garbage collector roots...
determining live/dead paths...
deleting '/nix/store/tmp-531510-1'
deleting '/nix/store/tmp-3358351-801'
2024-07-23 14:43:02,061:INFO:nix_heuristic_gc.graph: topologically sorting paths
2024-07-23 14:43:24,055:INFO:nix_heuristic_gc.graph: building graph
2024-07-23 14:43:51,192:INFO:nix_heuristic_gc.graph: populating output-drv or drv-output edges
Traceback (most recent call last):
  File "/nix/store/i1ivf8p4aiixylbbgbynzs1djsr86yc9-python3.11-nix-heuristic-gc-0.5.1/bin/.nix-heuristic-gc-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/i1ivf8p4aiixylbbgbynzs1djsr86yc9-python3.11-nix-heuristic-gc-0.5.1/lib/python3.11/site-packages/nix_heuristic_gc/__main__.py", line 184, in main
    nix_heuristic_gc(**parsed)
  File "/nix/store/i1ivf8p4aiixylbbgbynzs1djsr86yc9-python3.11-nix-heuristic-gc-0.5.1/lib/python3.11/site-packages/nix_heuristic_gc/__init__.py", line 49, in nix_heuristic_gc
    garbage_graph = GarbageGraph(
                    ^^^^^^^^^^^^^
  File "/nix/store/i1ivf8p4aiixylbbgbynzs1djsr86yc9-python3.11-nix-heuristic-gc-0.5.1/lib/python3.11/site-packages/nix_heuristic_gc/graph.py", line 216, in __init__
    for output in self.store.query_derivation_outputs(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: error: cannot operate on output 'out' of the unbuilt derivation '/nix/store/14cps7lngp7ybqd5hzngsnzxqiz3b9zr-nix-shell-env.drv'
```

**Comment by risicle** [comment]
Are you able to try the version I've put on my branch `ris-missing-realisation` ?

**Comment by DavHau** [comment]
Great, it worked with your branch now. Though I did execute a normal GC yesterday, so it might have removed the unbuilt derivation already. Anyways it seems to work now

**Comment by risicle** [comment]
Damnit, well let's assume this fixes it and anyone who finds otherwise can reopen.

**Comment by risicle** [comment]
Hopefully fixed by b78d5c799b46900d291e84edbb2fe9eacd858718
